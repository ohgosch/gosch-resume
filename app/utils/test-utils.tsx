import { render } from '@testing-library/react';
import i18n from 'i18next';
import { I18nContext } from 'next-i18next';
import en from 'public/locales/en/common.json';
import { type ReactElement } from 'react';
import { initReactI18next } from 'react-i18next';
import { ThemeProvider } from 'styled-components';

import theme from '../lib/visual/theme';

i18n.use(initReactI18next).init({
  lng: 'en',
  debug: false,
  ns: ['common', 'header', 'footer'],
  defaultNS: 'common',
  resources: {
    en: {
      common: en,
    },
  },
});

const customRender = (ui: ReactElement) =>
  render(
    <I18nContext.Provider value={{ i18n }}>
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </I18nContext.Provider>,
  );

export * from '@testing-library/react';
export { customRender as render };
