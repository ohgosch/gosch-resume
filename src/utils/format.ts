import { format as formatLib, parseISO } from 'date-fns';
import { enUS, ptBR } from 'date-fns/locale';

export const formatDate = (date = '', format = 'yyyy-MM-dd', locale = 'en') => {
  if (!date) return date;
  return formatLib(parseISO(date), format, {
    locale: locale === 'pt' ? ptBR : enUS,
  });
};
