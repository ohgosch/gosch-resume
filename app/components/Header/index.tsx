'use client';

import { type ISkeleton } from 'interfaces/resume';
import { EnvelopeIcon } from 'lib/statics/icons/EnvelopeIcon';
import { GithubIcon } from 'lib/statics/icons/GithubIcon';
import { LinkedinIcon } from 'lib/statics/icons/LinkedinIcon';
import { PrintIcon } from 'lib/statics/icons/PrintIcon';
import { SoundIcon } from 'lib/statics/icons/SoundIcon';
import { WhatsAppIcon } from 'lib/statics/icons/WhatsAppIcon';
import { LogoImage } from 'lib/statics/LogoImage';
import { useI18n, useScopedI18n } from 'locales/client';
import { useCallback, useMemo, useRef } from 'react';
import { formatPhone } from 'utils/format';

import * as S from './styles';

export interface HeaderProps {
  forceBorder?: boolean;
  skeleton: ISkeleton;
}

const Header = ({ forceBorder, skeleton: { attributes } }: HeaderProps) => {
  /*
   * Store's
   * */
  const scopedT = useScopedI18n('header');
  const t = useI18n();

  /*
   * Ref's
   * */
  const playerRef = useRef<HTMLAudioElement>(null);

  /*
   * State's
   * */

  /*
   * Memo's
   * */
  const phoneMasked = useMemo(
    () => formatPhone(attributes.phone),
    [attributes.phone],
  );

  /*
   * Callback's
   * */
  const triggerPlayAudio = useCallback(() => {
    void playerRef?.current?.play();
  }, []);

  /*
   * Effect's
   * */

  return (
    <S.Container forceBorder={forceBorder}>
      <S.Title>{t('common.name')}</S.Title>
      <S.Content>
        <S.LogoWrapper>
          <LogoImage />
        </S.LogoWrapper>

        <S.Menu aria-label={scopedT('social-media')}>
          <S.MenuItem aria-label="Linked-in">
            <S.MenuItemIconWrapper>
              <LinkedinIcon aria-hidden />
              <S.MenuItemText
                href={`https://linkedin.com/in/${attributes.linkedin}`}
                target="_blank"
                title="Linked-in"
              >
                @{attributes.linkedin}
              </S.MenuItemText>
            </S.MenuItemIconWrapper>
          </S.MenuItem>
          <S.MenuItem aria-label="Github">
            <S.MenuItemIconWrapper>
              <GithubIcon aria-hidden />
              <S.MenuItemText
                href={`https://github.com/${attributes.github}`}
                target="_blank"
                title="GitHub"
              >
                @{attributes.github}
              </S.MenuItemText>
            </S.MenuItemIconWrapper>
          </S.MenuItem>
          <S.MenuItem aria-label="WhatsApp">
            <S.MenuItemIconWrapper>
              <WhatsAppIcon aria-hidden />
              <S.MenuItemText
                href={`https://wa.me/${attributes.phone}`}
                target="_blank"
                title="WhatsApp"
              >
                {phoneMasked}
              </S.MenuItemText>
            </S.MenuItemIconWrapper>
          </S.MenuItem>
          {!!attributes.email && (
            <S.MenuItem aria-label={scopedT('mail')}>
              <S.MenuItemIconWrapper>
                <EnvelopeIcon aria-hidden />
                <S.MenuItemText
                  href={`mailto:${attributes.email}`}
                  target="_blank"
                  title={scopedT('mail')}
                >
                  {attributes.email}
                </S.MenuItemText>
              </S.MenuItemIconWrapper>
            </S.MenuItem>
          )}
        </S.Menu>
        <S.Infos>
          <S.Info
            onClick={() => {
              window.print();
            }}
            $hideInMobile
          >
            <PrintIcon />
            {scopedT('print')}
          </S.Info>
          <S.Info onClick={triggerPlayAudio}>
            <SoundIcon />
            {scopedT('listen-my-name')}
          </S.Info>
          <S.Player id="player" ref={playerRef}>
            <source src="/assets/audio/gosch.mp3" type="audio/mpeg" />
          </S.Player>
        </S.Infos>
      </S.Content>
    </S.Container>
  );
};

export default Header;
