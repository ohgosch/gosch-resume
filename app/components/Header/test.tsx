import { render, screen } from '@testing-library/react';
import Header, { type HeaderProps } from 'components/Header/index';

describe('<Header />', () => {
  const props: HeaderProps = {
    socialMedias: [
      {
        url: 'http://linked-in.com',
        display: 'Linked-in',
        slug: 'linkedin',
      },
      {
        url: 'http://linked-in.com',
        display: 'GitHub',
        slug: 'github',
      },
      {
        url: 'http://linked-in.com',
        display: 'WhatsApp',
        slug: 'whatsapp',
      },
    ],
  };

  it('should render ...', () => {
    render(<Header {...props} />);

    expect(screen.getByRole('heading', { name: /Gosch/i })).toBeInTheDocument();

    expect(screen.getByText(props.socialMedias[0].display)).toBeInTheDocument();
    expect(screen.getByText(props.socialMedias[1].display)).toBeInTheDocument();
    expect(screen.getByText(props.socialMedias[2].display)).toBeInTheDocument();
  });
});
