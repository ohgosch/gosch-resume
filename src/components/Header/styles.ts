import { hideVisually, rem, size } from 'polished';
import styled, { css } from 'styled-components';

import { mobile, print, tabletDesktop } from 'visual/medias';
import { HeaderProps } from 'components/Header/index';
import Wrapper from 'visual/styles/Wrapper';

const menuSizeTabletDesktop = 64;

type ContainerProps = Pick<HeaderProps, 'forceBorder'>;

export const Container = styled.header<ContainerProps>`
  ${({ theme, forceBorder }) => css`
    background-color: ${theme.colors.primary};

    ${mobile(css`
      border-bottom: ${rem(5)} solid ${theme.colors.secondary};
    `)}

    ${print(css`
      border-bottom: ${rem(5)} solid ${theme.colors.secondary};
    `)}

    ${forceBorder &&
    css`
      border-bottom: ${rem(5)} solid ${theme.colors.secondary};
    `}
  `}
`;

export const Title = styled.h1`
  ${hideVisually()};
`;

export const Content = styled(Wrapper)`
  position: relative;
  display: grid;
  align-items: center;
  justify-content: space-between;

  ${tabletDesktop(css`
    grid-template-areas: 'logo menu' 'infos menu';
  `)}

  ${mobile(css`
    min-height: ${rem(300)};
    grid-template-areas: 'logo logo' 'menu infos';
    align-items: end;
  `)}
  
  ${print(css`
    grid-template-areas: 'logo menu';
  `)}

  #logo {
    ${mobile(css`
      max-width: ${rem(300)};
    `)}

    ${tabletDesktop(css`
      width: ${rem(330)};
    `)}
  }
`;

export const LogoWrapper = styled.div`
  grid-area: logo;

  ${tabletDesktop(
    css`
      display: flex;
      align-items: end;
      height: 100%;
    `,
    true,
  )}

  ${mobile(css`
    width: 100%;
    display: block;
    text-align: center;
    justify-content: center;
    margin-bottom: ${rem(-72)};
  `)}
  
  ${print(css`
    height: 100%;
    display: flex;
    align-items: center;
  `)}
`;

export const Infos = styled.div`
  grid-area: infos;
  display: flex;
  gap: ${rem(16)};
  align-items: end;

  ${mobile(css`
    height: ${rem(36)};
    margin-bottom: ${rem(24)};
    justify-content: end;
    align-items: center;
  `)}

  ${print(css`
    display: none;
  `)}

    svg {
    ${size(rem(12))}
  }
`;

type InfoProps = {
  hideInMobile?: boolean;
};

export const Info = styled.button<InfoProps>`
  ${({ theme, hideInMobile }) => css`
    background-color: transparent;
    color: ${theme.colors.white};
    display: flex;
    gap: ${rem(4)};
    width: fit-content;
    font-style: normal;
    font-weight: 700;
    font-size: ${rem(12)};
    line-height: ${rem(15)};
    text-transform: lowercase;
    align-items: center;
    padding: ${rem(4)};

    ${mobile(css`
      ${hideInMobile &&
      css`
        display: none;
      `}
    `)}
  `}
`;

export const Menu = styled.ul`
  ${({ theme }) => css`
    list-style-type: none;
    display: flex;
    grid-area: menu;

    ${tabletDesktop(css`
      width: ${rem(menuSizeTabletDesktop)};
      flex-direction: column;
      background-color: ${theme.colors.secondary};
      justify-content: center;
      padding: ${rem(22)} 0;
    `)}

    ${mobile(css`
      gap: ${rem(10)};
      position: absolute;
      bottom: ${rem(24)};
    `)}
  `}
`;

export const MenuItemIconWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;

    .icon {
      ${tabletDesktop(css`
        position: relative;
        z-index: 10;
        pointer-events: none;
        ${size(rem(28))}
      `)}

      ${mobile(css`
        ${size(rem(16))}

        path {
          fill: ${theme.colors.secondary};
        }
      `)}
    }
  `}
`;

export const MenuItemText = styled.a`
  ${({ theme }) => css`
    ${tabletDesktop(css`
      font-weight: 400;
      font-size: ${rem(24)};
      line-height: ${rem(menuSizeTabletDesktop)};
      color: ${theme.colors.secondary};
      text-decoration: none;
      position: absolute;
      transition: 0.2s ease;
      right: 0;
      padding-right: ${rem(menuSizeTabletDesktop + 24)};
      padding-left: ${rem(24)};
      border-bottom-left-radius: ${rem(5)};
      border-top-left-radius: ${rem(5)};
      height: ${rem(menuSizeTabletDesktop)};
      white-space: nowrap;
    `)}

    ${mobile(css`
      font-size: 0;
      width: 100%;
      height: 100%;
      position: absolute;
      left: 0;
      top: 0;
    `)}
  `}
`;

export const MenuItem = styled.li`
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;

  ${({ theme }) => css`
    ${tabletDesktop(css`
      ${size(rem(menuSizeTabletDesktop))}

      :hover ${MenuItemText} {
        background-color: ${theme.colors.secondary};
        color: ${theme.colors.primary};
      }
    `)}

    ${mobile(css`
      ${size(rem(36))};
      border: ${rem(2)} solid ${theme.colors.secondary};
      border-radius: ${rem(5)};
    `)}
  `}
`;

export const Player = styled.audio(hideVisually);
