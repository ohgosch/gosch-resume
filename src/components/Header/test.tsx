import { render, screen } from 'utils/test-utils';

import Header, { HeaderProps } from '.';

describe('<Header />', () => {
  const props: HeaderProps = {
    data: [
      {
        url: 'http://linked-in.com',
        display: 'Linked-in',
        slug: 'linkedin',
      },
      {
        url: 'http://linked-in.com',
        display: 'GitHub',
        slug: 'github',
      },
      {
        url: 'http://linked-in.com',
        display: 'WhatsApp',
        slug: 'whatsapp',
      },
    ],
  };

  it('should render ...', () => {
    render(<Header {...props} />);

    expect(screen.getByRole('heading', { name: /Gosch/i })).toBeInTheDocument();

    expect(screen.getByText(props.data[0].display)).toBeInTheDocument();
    expect(screen.getByText(props.data[1].display)).toBeInTheDocument();
    expect(screen.getByText(props.data[2].display)).toBeInTheDocument();
  });
});
