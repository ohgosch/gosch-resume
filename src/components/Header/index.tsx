import { useTranslation, withTranslation } from 'next-i18next';

import WhatsAppIcon from 'public/assets/img/icon/whatsapp.svg';
import LinkedInIcon from 'public/assets/img/icon/linkedin.svg';
import GitHubIcon from 'public/assets/img/icon/github.svg';
import Logo from 'public/assets/img/logo.svg';

import * as S from './styles';

const Header = () => {
  const { t } = useTranslation(['header', 'common']);

  return (
    <S.Container>
      <S.Title>{t('name', { ns: 'common' })}</S.Title>
      <S.Content>
        <Logo id="logo" aria-label={t('logo-alt')} />
        <S.Menu aria-label={t('social-media')}>
          <S.MenuItem aria-label="Linked-in">
            <S.MenuItemIconWrapper>
              <LinkedInIcon className="icon" aria-hidden />
              <S.MenuItemText href="/" title="Linked-in">
                @ohgosch
              </S.MenuItemText>
            </S.MenuItemIconWrapper>
          </S.MenuItem>
          <S.MenuItem aria-label="Github">
            <S.MenuItemIconWrapper>
              <GitHubIcon className="icon" aria-hidden />
              <S.MenuItemText href="/" title="GitHub">
                @ohgosch
              </S.MenuItemText>
            </S.MenuItemIconWrapper>
          </S.MenuItem>
          <S.MenuItem aria-label="WhatsApp">
            <S.MenuItemIconWrapper>
              <WhatsAppIcon className="icon" aria-hidden />
              <S.MenuItemText href="/" title="WhatsApp">
                +XX XX XXXXX-XXXX
              </S.MenuItemText>
            </S.MenuItemIconWrapper>
          </S.MenuItem>
        </S.Menu>
      </S.Content>
    </S.Container>
  );
};

export default withTranslation()(Header);
