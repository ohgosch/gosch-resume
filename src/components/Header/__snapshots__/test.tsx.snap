// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render ... 1`] = `
.c2 {
  margin-left: auto;
  margin-right: auto;
}

.c0 {
  background-color: #63415F;
}

.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (max-width:46rem) {
  .c2 {
    width: 90%;
    max-width: 46rem;
  }
}

@media (min-width:46.0625rem) and (max-width:64rem) {
  .c2 {
    width: 88%;
    max-width: 64rem;
  }
}

@media (min-width:64.0625rem) {
  .c2 {
    width: 90%;
    max-width: 64.0625rem;
  }
}

@media (max-width:46rem) {
  .c0 {
    border-bottom: 0.3125rem solid #F2E6DF;
  }
}

@media (min-width:46.0625rem) {
  .c3 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (max-width:46rem) {
  .c3 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    min-height: 18.75rem;
  }
}

@media (max-width:46rem) {
  .c3 #logo {
    max-width: 18.75rem;
  }
}

@media (min-width:46.0625rem) {
  .c3 #logo {
    width: 20.625rem;
  }
}

@media (min-width:46.0625rem) {
  .c4 {
    width: 4rem;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    background-color: #F2E6DF;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    padding: 3.375rem 0;
  }
}

@media (max-width:46rem) {
  .c4 {
    gap: 0.625rem;
    position: absolute;
    bottom: 1.5rem;
  }
}

@media (min-width:46.0625rem) {
  .c5 {
    height: 4rem;
    width: 4rem;
  }
}

@media (max-width:46rem) {
  .c5 {
    height: 2.25rem;
    width: 2.25rem;
    border: 0.125rem solid #F2E6DF;
    border-radius: 0.3125rem;
  }
}

@media (min-width:46.0625rem) {
  .c6 .icon {
    height: 1.75rem;
    width: 1.75rem;
  }
}

@media (max-width:46rem) {
  .c6 .icon {
    height: 1rem;
    width: 1rem;
  }

  .c6 .icon path {
    fill: #F2E6DF;
  }
}

@media (min-width:46.0625rem) {
  .c7 {
    font-weight: 400;
    font-size: 1.5rem;
    line-height: 4rem;
    color: #F2E6DF;
    -webkit-text-decoration: none;
    text-decoration: none;
    position: absolute;
    right: 0;
    padding-right: 5.5rem;
    height: 4rem;
    white-space: nowrap;
  }
}

@media (max-width:46rem) {
  .c7 {
    font-size: 0;
    width: 100%;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0;
  }
}

<header
  class="c0"
>
  <h1
    class="c1"
  >
    Gosch
  </h1>
  <div
    class="c2 c3"
  >
    <svg
      aria-label="logo-alt"
      fill="none"
      id="logo"
      viewBox="0 0 330 102"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M120.598 66.27a22.776 22.776 0 0 1-8.282-8.362 22.978 22.978 0 0 1-3.03-11.408c0-4.005 1.045-7.939 3.03-11.409a22.776 22.776 0 0 1 8.282-8.362 22.178 22.178 0 0 1 11.398-3.048c.941 0 1.882.063 2.815.192a1.194 1.194 0 0 1 .871.507 1.224 1.224 0 0 1 .187.995l-1.005 5.477a1.132 1.132 0 0 1-.809.967c-.143.04-.293.053-.44.035l-1.619-.064a14.14 14.14 0 0 0-7.303 1.982 14.641 14.641 0 0 0-5.291 5.328 14.728 14.728 0 0 0 0 14.715 14.566 14.566 0 0 0 5.291 5.327 14.14 14.14 0 0 0 7.303 1.982 12.79 12.79 0 0 0 7.99-2.61 15.146 15.146 0 0 0 5.058-6.948h-8.053a1.178 1.178 0 0 1-.879-.373 1.245 1.245 0 0 1-.37-.884v-4.582c.005-.332.139-.648.372-.883.233-.234.547-.369.877-.374h13.303a4.454 4.454 0 0 1 2.307.5 1.627 1.627 0 0 1 .815 1.45 24.294 24.294 0 0 1-2.752 11.497 21.115 21.115 0 0 1-7.651 8.268 20.396 20.396 0 0 1-11.017 3.048 22.179 22.179 0 0 1-11.398-2.962zm45.581-4.645a22.525 22.525 0 0 1-4.53-7.736l-.116-.32a22.136 22.136 0 0 1-1.132-7.032 23.353 23.353 0 0 1 1.312-7.672 21.397 21.397 0 0 1 3.937-6.851 29.031 29.031 0 0 1 6.963-5.69 20.373 20.373 0 0 1 8.901-2.27h.18a21.882 21.882 0 0 1 10.244 1.876 22.38 22.38 0 0 1 6.022 4.06 18.737 18.737 0 0 1 4.233 5.327 22.287 22.287 0 0 1 2.963 11.113 22.986 22.986 0 0 1-1.058 6.884 22.58 22.58 0 0 1-3.175 6.255 23.71 23.71 0 0 1-3.397 3.91 23.3 23.3 0 0 1-4.53 3.111 21.965 21.965 0 0 1-14.254 2.193 22.06 22.06 0 0 1-12.563-7.126v-.032zm27.198-5.605a14.378 14.378 0 0 0 2.435-3.719 14.81 14.81 0 0 0 1.248-5.87 14.406 14.406 0 0 0-2.254-7.768 13.647 13.647 0 0 0-6.18-5.328 14.6 14.6 0 0 0-5.927-1.257 14.146 14.146 0 0 0-5.291 1.001 13.889 13.889 0 0 0-4.53 2.952l-1.058 1.14a14.858 14.858 0 0 0-2.53 4.454 14.814 14.814 0 0 0 .466 10.975 13.259 13.259 0 0 0 3.81 4.933 14.32 14.32 0 0 0 15.716 1.854 15.646 15.646 0 0 0 4.095-3.367zm25.4 10.656a19.304 19.304 0 0 1-7.091-3.453 1.307 1.307 0 0 1-.37-.948c.043-.554.213-1.09.497-1.566a27.818 27.818 0 0 1 1.662-2.558 1.287 1.287 0 0 1 1.058-.628c.192.015.381.058.561.128a25.664 25.664 0 0 0 13.483 3.452c1.74.1 3.475-.268 5.027-1.066a3.276 3.276 0 0 0 1.346-1.208 3.306 3.306 0 0 0 .506-1.743 5.348 5.348 0 0 0-1.312-3.197 5.907 5.907 0 0 0-2.593-1.79 42.938 42.938 0 0 0-4.456-1.481c-.211 0-.751-.192-1.63-.437a25.605 25.605 0 0 1-7.863-3.612 10.036 10.036 0 0 1-3.746-5.328 13.325 13.325 0 0 1-.688-4.027c-.016-2.08.569-4.12 1.683-5.872a13.093 13.093 0 0 1 4.868-4.432 19.572 19.572 0 0 1 7.493-2.206c.381 0 .963-.053 1.756-.053 3.188.006 6.346.624 9.303 1.822 3.238 1.634 4.674 3.015 4.307 4.145a10.52 10.52 0 0 1-1.439 2.333c-.73.949-1.196 1.567-1.408 1.854a.634.634 0 0 1-.561.256 5.814 5.814 0 0 1-2.116-.756 14.753 14.753 0 0 0-3.323-1.194 23.142 23.142 0 0 0-4.689-.628 12.76 12.76 0 0 0-4.053.692 7.307 7.307 0 0 0-2.497 1.918 3.669 3.669 0 0 0-.879 2.291 3.777 3.777 0 0 0 1.757 3.08 21.1 21.1 0 0 0 4.149 2.333c1.471.629 3.174 1.279 4.963 1.95 1.997.71 3.609 1.389 4.836 2.035a8.654 8.654 0 0 1 3.027 2.419 9.117 9.117 0 0 1 2.117 3.399c.544 1.355.831 2.8.846 4.262a10.11 10.11 0 0 1-1.873 6.106 12.116 12.116 0 0 1-5.185 3.985 19.15 19.15 0 0 1-7.62 1.417 32.603 32.603 0 0 1-9.874-1.63l.021-.064zm42.766-.682a22.655 22.655 0 0 1-8.086-8.299 22.852 22.852 0 0 1 0-22.477 22.656 22.656 0 0 1 8.086-8.298 21.174 21.174 0 0 1 11.091-3.047c3.073-.01 6.113.632 8.922 1.886a21.347 21.347 0 0 1 7.408 5.402c.112.106.203.232.267.373.063.141.098.293.103.448a1.234 1.234 0 0 1-.434.81 71.84 71.84 0 0 0-2.624 2.237l-1.81 1.598a1.193 1.193 0 0 1-.878.384 1.146 1.146 0 0 1-.815-.383 13.611 13.611 0 0 0-4.577-3.34 13.519 13.519 0 0 0-5.53-1.178 13.163 13.163 0 0 0-6.964 1.939 14.376 14.376 0 0 0-5.09 5.221 14.632 14.632 0 0 0-1.873 7.166c0 2.511.645 4.98 1.873 7.165a14.28 14.28 0 0 0 5.09 5.222 13.174 13.174 0 0 0 6.964 1.95c1.779.003 3.54-.36 5.175-1.066a14.12 14.12 0 0 0 4.498-3.09 2.173 2.173 0 0 1 1.376-.5 1.313 1.313 0 0 1 .995.383c.423.33 1.005.82 1.756 1.47.752.65 1.387 1.226 1.926 1.737a1.685 1.685 0 0 1 .688 1.375 1.187 1.187 0 0 1-.37.884 21.901 21.901 0 0 1-12.866 6.807 21.823 21.823 0 0 1-14.269-2.78h-.032zm37.126 2.195a.837.837 0 0 1-.254-.629V25.311a.863.863 0 0 1 .254-.628.859.859 0 0 1 .624-.245h6.551a.8.8 0 0 1 .624.277c.156.16.247.373.254.596v16.41c.011.38.166.743.434 1.012.267.27.627.426 1.006.437h11.863a1.333 1.333 0 0 0 .963-.405c.28-.286.445-.665.466-1.066V25.312a.9.9 0 0 1 .254-.597.813.813 0 0 1 .624-.277h6.551a.842.842 0 0 1 .818.533.861.861 0 0 1 .061.34v42.313a.849.849 0 0 1-.254.629.836.836 0 0 1-.625.255h-6.572a.786.786 0 0 1-.624-.287.884.884 0 0 1-.254-.597V53.037c-.021-.4-.186-.78-.466-1.066a1.305 1.305 0 0 0-.963-.405H308.14a1.36 1.36 0 0 0-.973.373 1.56 1.56 0 0 0-.466 1.002v14.683a.884.884 0 0 1-.254.597.78.78 0 0 1-.624.287h-6.551a.831.831 0 0 1-.625-.255l.022-.064zM94.217 50.054V6.428a1.855 1.855 0 0 0-.553-1.358 1.832 1.832 0 0 0-1.363-.522h-14.02a1.686 1.686 0 0 0-1.332.599 1.91 1.91 0 0 0-.538 1.28v35.038a3.534 3.534 0 0 1-1.004 2.22 2.827 2.827 0 0 1-2.068.87H59.143a2.75 2.75 0 0 0-1.869.811 2.78 2.78 0 0 0-.807 1.88v9.799c.021.707.31 1.38.807 1.88.497.5 1.166.79 1.87.81h17.22a32.524 32.524 0 0 1-10.818 14.84 27.47 27.47 0 0 1-17.082 5.57 30.317 30.317 0 0 1-15.62-4.23 31.273 31.273 0 0 1-11.416-11.48c-2.755-4.77-4.206-10.187-4.206-15.703s1.451-10.934 4.206-15.703a31.272 31.272 0 0 1 11.415-11.48 30.317 30.317 0 0 1 15.621-4.23l3.505.13a2.41 2.41 0 0 0 1.76-.543c.491-.407.804-.993.87-1.631l2.137-11.69a2.612 2.612 0 0 0-.413-2.112A2.586 2.586 0 0 0 54.47.4a43.942 43.942 0 0 0-6.006-.4 47.995 47.995 0 0 0-18.583 3.647 48.213 48.213 0 0 0-15.755 10.559 48.523 48.523 0 0 0-10.5 15.844A48.742 48.742 0 0 0 .002 48.738a48.714 48.714 0 0 0 6.398 24.367 48.368 48.368 0 0 0 17.693 17.85 47.383 47.383 0 0 0 24.372 6.567 43.664 43.664 0 0 0 23.554-6.51A45.088 45.088 0 0 0 88.375 73.39a51.398 51.398 0 0 0 5.842-22.865v-.47z"
        fill="#F2E6DF"
      />
    </svg>
    <ul
      aria-label="social-media"
      class="c4"
    >
      <li
        aria-label="Linked-in"
        class="c5"
      >
        <div
          class="c6"
        >
          <svg
            aria-hidden="true"
            class="icon"
            fill="none"
            viewBox="0 0 27 27"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.359 26.5V8.98H.91V26.5h5.449zM3.605 6.637c1.758 0 3.164-1.465 3.164-3.223C6.77 1.715 5.363.309 3.605.309A3.13 3.13 0 0 0 .5 3.414c0 1.758 1.406 3.223 3.105 3.223zM26.691 26.5h.059v-9.609c0-4.687-1.055-8.32-6.562-8.32-2.637 0-4.395 1.465-5.156 2.812h-.059V8.98H9.758V26.5h5.449v-8.672c0-2.285.41-4.453 3.223-4.453s2.871 2.578 2.871 4.629V26.5h5.391z"
              fill="#63415f"
            />
          </svg>
          <a
            class="c7"
            href="/"
            title="Linked-in"
          >
            @ohgosch
          </a>
        </div>
      </li>
      <li
        aria-label="Github"
        class="c5"
      >
        <div
          class="c6"
        >
          <svg
            aria-hidden="true"
            class="icon"
            fill="none"
            viewBox="0 0 30 30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.668 23.57c0-.117-.117-.234-.293-.234s-.293.117-.293.234.117.234.293.176c.176 0 .293-.059.293-.176zm-1.816-.293c0 .117.117.293.293.293.117.059.293 0 .352-.117 0-.117-.059-.234-.234-.293s-.352 0-.41.117zm2.637-.058c-.176 0-.293.117-.293.293 0 .117.176.176.352.117s.293-.117.234-.234c0-.117-.176-.234-.293-.176zm3.809-22.5C6.211.719 0 6.93 0 15.016c0 6.504 4.043 12.07 9.902 14.063.762.117.996-.351.996-.703v-3.633s-4.102.879-4.98-1.758c0 0-.645-1.699-1.582-2.109 0 0-1.348-.937.059-.937a3.14 3.14 0 0 0 2.285 1.523c1.289 2.285 3.398 1.641 4.277 1.231.117-.937.469-1.582.938-1.992-3.281-.352-6.621-.82-6.621-6.445 0-1.641.469-2.402 1.406-3.457-.176-.41-.645-1.934.176-3.984 1.172-.352 4.043 1.582 4.043 1.582 1.172-.352 2.402-.469 3.633-.469 1.289 0 2.52.117 3.691.469 0 0 2.813-1.992 4.043-1.582.82 2.051.293 3.574.176 3.984.938 1.055 1.523 1.816 1.523 3.457 0 5.625-3.457 6.094-6.738 6.445.527.469.996 1.348.996 2.754l-.059 4.863c0 .41.293.879 1.055.703 5.859-1.934 9.844-7.5 9.844-14.004C29.063 6.93 22.441.719 14.297.719zM5.684 20.934c-.117.059-.059.234 0 .352.117.059.234.117.352.059.059-.059.059-.234-.059-.352-.117-.059-.234-.117-.293-.059zm-.645-.469c-.059.117 0 .176.117.234s.234.059.293-.059c0-.059-.059-.117-.176-.176s-.176-.059-.234 0zm1.875 2.109c-.059.059-.059.234.117.352.117.117.293.176.352.059.059-.059.059-.234-.059-.352s-.293-.176-.41-.059zm-.645-.879c-.117.059-.117.234 0 .352s.234.176.352.117c.059-.059.059-.234 0-.352-.117-.117-.234-.176-.352-.117z"
              fill="#63415f"
            />
          </svg>
          <a
            class="c7"
            href="/"
            title="GitHub"
          >
            @ohgosch
          </a>
        </div>
      </li>
      <li
        aria-label="WhatsApp"
        class="c5"
      >
        <div
          class="c6"
        >
          <svg
            aria-hidden="true"
            class="icon"
            fill="none"
            viewBox="0 0 27 27"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M22.766 3.992a12.8 12.8 0 0 0-9.2-3.867C6.418.125.559 5.984.559 13.133c0 2.344.645 4.57 1.758 6.504L.5 26.375l6.855-1.758c1.934.996 4.043 1.582 6.211 1.582 7.207 0 13.184-5.859 13.184-13.008 0-3.516-1.523-6.738-3.984-9.199zm-9.199 19.981c-1.934 0-3.809-.527-5.508-1.523l-.352-.234-4.102 1.113 1.113-3.984-.293-.41c-1.055-1.758-1.641-3.75-1.641-5.801 0-5.918 4.863-10.781 10.84-10.781a10.68 10.68 0 0 1 7.617 3.164c2.051 2.051 3.281 4.746 3.281 7.676 0 5.918-4.98 10.781-10.957 10.781zm5.977-8.086c-.352-.176-1.934-.937-2.227-1.055s-.527-.176-.762.176c-.176.293-.82 1.055-.996 1.289-.234.176-.41.234-.703.059-1.934-.937-3.164-1.699-4.453-3.867-.351-.586.352-.527.938-1.758.117-.234.059-.41 0-.586s-.762-1.758-.996-2.402c-.293-.645-.527-.586-.762-.586h-.586a1.1 1.1 0 0 0-.879.41c-.293.352-1.113 1.113-1.113 2.695 0 1.641 1.113 3.164 1.289 3.399.176.176 2.285 3.457 5.566 4.863 2.051.938 2.871.996 3.926.82.586-.059 1.875-.762 2.168-1.524s.293-1.406.176-1.523c-.059-.176-.293-.234-.586-.41z"
              fill="#63415f"
            />
          </svg>
          <a
            class="c7"
            href="/"
            title="WhatsApp"
          >
            +XX XX XXXXX-XXXX
          </a>
        </div>
      </li>
    </ul>
  </div>
</header>
`;
